### PR TITLE
Stop a security analysis computation in progress

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
When the security-analysis-server is stopped or no started and we request the launch of an AS we have no feedback

File: src/components/study-pane.js [hunk 1]
```diff
@@ -307,6 +310,8 @@ const StudyPane = (props) => {
         // close the contingency list selection window
         setShowContingencyListSelector(false);
 
+        setComputationStopped(false);
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:313]: See comment about status `stopped`

File: src/components/study-pane.js [hunk 2]
```diff
@@ -322,6 +327,13 @@ const StudyPane = (props) => {
         }
     };
 
+    const stopComputation = (runnable) => {
+        if (runnable === Runnable.SECURITY_ANALYSIS) {
+            stopSecurityAnalysis(studyName, userId);
+            setComputationStopped(!computationStopped);
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:333]: See comment about status `stopped`

File: src/components/study-pane.js [hunk 3]
```diff
@@ -974,10 +986,15 @@ const StudyPane = (props) => {
                     >
                         <RunButton
                             runnables={RUNNABLES}
+                            stopComputationText={[
+                                intl.formatMessage({ id: 'StopComputation' }),
+                            ]}
```
Comment (SlimaneAmar) [inline src/components/study-pane.js:991]: See comment about enum `actionsOnRunnable`

File: src/components/util/split-button.js [hunk 4]
```diff
@@ -68,8 +71,14 @@ const SplitButton = (props) => {
     };
 
     const handleMenuItemClick = (event, index) => {
-        if (props.onSelectionChange) {
-            props.onSelectionChange(index);
+        if (!props.isRunning) {
+            if (props.onSelectionChange) {
```
Comment (SlimaneAmar) [inline src/components/util/split-button.js:75]: Is it not required ?

File: src/components/util/split-button.js [hunk 5]
```diff
@@ -68,8 +71,14 @@ const SplitButton = (props) => {
...
+                props.onSelectionChange(index);
+            }
+        } else {
+            if (props.onStopComputation) {
```
Comment (SlimaneAmar) [inline src/components/util/split-button.js:79]: Is it not required ?

File: src/components/util/split-button.js [hunk 6]
```diff
@@ -154,7 +166,14 @@ const SplitButton = (props) => {
                                                 )
                                             }
                                         >
-                                            {breakText(option)}
+                                            {props.isRunning && (
+                                                <ListItemIcon>
+                                                    <StopIcon fontSize="small" />
+                                                </ListItemIcon>
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/util/split-button.js:174]: color changed

File: src/components/run-button.js [hunk 7]
```diff
@@ -152,6 +184,8 @@ RunButton.propTypes = {
     getText: PropTypes.func.isRequired,
     getStartIcon: PropTypes.func.isRequired,
     onStartClick: PropTypes.func,
+    stopComputationText: PropTypes.array.isRequired,
+    computationStopped: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/run-button.js]: isRequired added

**Review by ChamseddineBhd — APPROVED** [review]

File: src/components/run-button.js [hunk 8]
```diff
@@ -121,27 +144,32 @@ const RunButton = (props) => {
         return props.getStatus(getRunnable());
     }
 
-    const runningStatus = getRunningStatus();
```
Comment (jonenst) [inline src/components/run-button.js:124]: why remove this ?

File: src/components/study-pane.js (hunk 2 above)
Comment (jonenst) [reply src/components/study-pane.js:333]: does this really need to be a toggle ? why not setComputationStopped(true) ?

File: src/components/util/split-button.js [hunk 9]
```diff
@@ -68,8 +74,10 @@ const SplitButton = (props) => {
     };
 
     const handleMenuItemClick = (event, index) => {
-        if (props.onSelectionChange) {
+        if (!props.isRunning) {
```
Comment (jonenst) [inline src/components/util/split-button.js:77]: why remove the check for onSelectionChange ?

File: src/translations/fr.json [hunk 10]
```diff
@@ -210,5 +210,7 @@
     "NominalV": "Tension nominale (kV)",
     "HvdcLineId": "ID ligne HVDC",
     "PowerFactor": "Facteur de puissance",
-    "LossFactor": "Facteur de perte"
+    "LossFactor": "Facteur de perte",
+
+    "StopComputation" : "Arr\u00eater le calcul"
```
Comment (jonenst) [inline src/translations/fr.json:215]: why escape ?

File: src/components/run-button.js [hunk 11]
```diff
@@ -121,27 +144,32 @@ const RunButton = (props) => {
...
             buttonDisabled={disabled}
-            selectionDisabled={runningStatus === RunningStatus.RUNNING}
+            selectionDisabled={selectedIndex === 0 && isRunning()}
             startIcon={props.getStartIcon(getRunningStatus())}
             text={
                 props.getText
                     ? props.getText(getRunnable(), getRunningStatus())
                     : ''
             }
+            onStopComputation={handleStopComputationClick}
```
Comment (jonenst) [inline src/components/run-button.js:170]: I think it's weird that runnableActions are generic but in contrast onStopComputation is not. Either make everything generic or everything specific

File: src/components/run-button.js (hunk 8 above)
Comment (FranckLecuyer) [reply src/components/run-button.js:124]: defining this variable is not really useful

File: src/components/util/split-button.js (hunk 9 above)
Comment (FranckLecuyer) [reply src/components/util/split-button.js:77]: property onSelectionChange is always defined. I will add required on this property

File: src/components/run-button.js (hunk 11 above)
Comment (FranckLecuyer) [reply src/components/run-button.js:170]: why is onStopComputation  not generic ?

Comment (FranckLecuyer) [reply src/components/run-button.js:170]: Done

Comment (jonenst) [reply src/components/run-button.js:170]: if you add a new runnableActions, for exemple "pause computation" or "restart computation", the onStopComputation callback is called (obviously you can't know what to call based only on the i18n label)

**Pull request merged**